nucliadb/nucliadb/search/api/v1/search.py

```diff
@@ -371,6 +372,9 @@ async def search(
     await abort_transaction()
 
     response.status_code = 206 if incomplete_results else 200
+    if check_missing_vectors(item, pb_query):
```

Comment:
hmm, okay since we don't have a great way of handling this, we're just assuming empty vector list translates to triton and then we do a 206

What do you think about pulling vectors into being a separate step here and not embedding into `global_query_to_pb`?

That way we can specifically handle the triton service issues instead of having to handle the error through a proxy object and config.

So maybe we'd only handle `SendToPredictError` in this function then?

Reply:
yes, thanks for the suggestions! Looks cleaner this way